x-pack/plugins/cloud_security_posture/server/routes/index.ts

```diff
@@ -5,18 +5,68 @@
  * 2.0.
  */
 
+import type { CoreSetup, KibanaRequest, Logger } from '@kbn/core/server';
+import type {
+  CspRequestHandlerContext,
+  CspServerPluginStart,
+  CspServerPluginStartDeps,
+} from '../types';
+import { PLUGIN_ID } from '../../common';
 import { defineGetComplianceDashboardRoute } from './compliance_dashboard/compliance_dashboard';
 import { defineGetBenchmarksRoute } from './benchmarks/benchmarks';
 import { defineUpdateRulesConfigRoute } from './configuration/update_rules_configuration';
 import { defineGetCspSetupStatusRoute } from './setup_status/setup_status';
 import { defineEsPitRoute } from './es_pit/es_pit';
-import { CspAppContext } from '../plugin';
-import { CspRouter } from '../types';
 
-export function defineRoutes(router: CspRouter, cspContext: CspAppContext) {
-  defineGetComplianceDashboardRoute(router, cspContext);
-  defineGetBenchmarksRoute(router, cspContext);
-  defineUpdateRulesConfigRoute(router, cspContext);
-  defineGetCspSetupStatusRoute(router, cspContext);
-  defineEsPitRoute(router, cspContext);
+export function setupRoutes({
+  core,
+  logger,
+}: {
+  core: CoreSetup<CspServerPluginStartDeps, CspServerPluginStart>;
+  logger: Logger;
+}) {
+  const router = core.http.createRouter<CspRequestHandlerContext>();
+  defineGetComplianceDashboardRoute(router);
+  defineGetBenchmarksRoute(router);
+  defineUpdateRulesConfigRoute(router);
+  defineGetCspSetupStatusRoute(router);
+  defineEsPitRoute(router);
+
+  core.http.registerRouteHandlerContext<CspRequestHandlerContext, typeof PLUGIN_ID>(
+    PLUGIN_ID,
+    (context, request) =>
+      createCspRouteContext({
+        core,
+        logger,
+        context,
+        request,
+      })
+  );
+}
```

Comment:
this function: 

1. registers routes
2. registers a route handler context, which is essentially a function that's called before every route's handler is called to provide its context. 

currently, we're not doing anything special there other than matching the previous `CspAppService`